streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollector.java

```diff
@@ -81,7 +81,9 @@ public RecordCollector(Producer<byte[], byte[]> producer) {
             if (partitions != null)
                 partition = partitioner.partition(record.key(), record.value(), partitions.size());
         }
-        this.producer.send(new ProducerRecord<>(record.topic(), partition, keyBytes, valBytes), callback);
+        this.producer.send(new ProducerRecord<>(record.topic(), partition, record.timestamp(),
```

Comment:
It looks like we could optimize this. I'll create a JIRA for it.

I'm not sure what you mean by changelogger?